src/main.cpp

```diff
@@ -0,0 +1,182 @@
+    auto method = phosphor::dbus::bus.new_method_call(
+        service.c_str(), SNMPManagerPath, SNMPManagerIface, "Client");
+    method.append(address, port);
+
+    phosphor::dbus::Path path;
+    try
+    {
+        phosphor::dbus::bus.call(method).read(path);
+    }
+    catch (const sdbusplus::exception::SdBusError& ex)
+    {
+        static constexpr auto InvalidArgument =
+            "xyz.openbmc_project.Common.Error.InvalidArgument";
+        if (0 == strcmp(ex.name(), InvalidArgument))
+        {
+            throw SNMPMgrError("Create SNMP trap receiver failed, %s",
+                               ex.description());
+        }
+        else
+        {
+            // NOTE: This is unlikely case.
+            throw SNMPMgrError("%s.Client('%s', %u) failed, %s",
+                               SNMPManagerIface, address.c_str(), port,
+                               ex.what());
+        }
+    }
+
+    printf("Receiver #%s added: address=%s, port=%d\n",
+           path.c_str() + SNMPMgrPathLen + 1, address.c_str(), port);
+}
+
+/**
+ * @brief Remove configured SNMP trap receiver
+ *
+ * @param index - Receiver index in the list
+ */
+void dropReceiver(size_t index)
+{
+    phosphor::dbus::Path path = SNMPManagerPath;
+    path += "/";
+    path += std::to_string(index);
+
+    try
+    {
+        auto service = phosphor::dbus::getService(path, SNMPReceiverIface);
+        auto method = phosphor::dbus::bus.new_method_call(
+            service.c_str(), path.c_str(), "xyz.openbmc_project.Object.Delete",
+            "Delete");
+        phosphor::dbus::bus.call_noreply(method);
+    }
+    catch (const ServiceNotFound&)
+    {
+        throw SNMPMgrError("SNMP trap receiver #%zu doesn't exist!", index);
+    }
+    catch (const sdbusplus::exception::SdBusError& ex)
+    {
+        // NOTE: This is unlikely case.
+        throw SNMPMgrError("Delete object at '%s' failed, %s", path.c_str(),
+                           ex.what());
+    }
+
+    printf("SNMP trap receiver #%zu removed!\n", index);
+}
+
+/**
+ * @brief Application entry point
+ *
+ * @param argc   - Number of command line arguments
+ * @param argv[] - Array of command line arguments
+ *
+ * @return - Return code
+ */
+int main(int argc, char* argv[])
+{
+    CLI::App app("SNMP trap receivers manager");
+
+#if REMOTE_HOST
+    app.add_option_function<std::string>(
+        "--host",
+        [](const std::string& host) {
+            if (!host.empty())
+            {
+                printf("Open D-Bus session on %s\n", host.c_str());
+                sd_bus* b = nullptr;
+                sd_bus_open_system_remote(&b, host.c_str());
+                phosphor::dbus::bus = sdbusplus::bus::bus(b, std::false_type());
+            }
+        },
+        "Operate on remote host");
+#endif
+
+    CLI::App* list = app.add_subcommand("list", "List configured receivers");
+    list->callback(listReceivers);
+
+    constexpr auto DefaultSNMPTrapPort = 162;
```

Comment:
Do you suggest to ask the external service for getting this well known constant?
It would look too overheaded.
`getservbyname` returns a pointer to the structure that has a port number in the network notation.
So, we would be forced to call `ntohs()` after.

Reply:
Well, that's just why `/etc/services` is there in the first place. Everyone could just hard-code those values, but then why keep that file? Not a big deal actually, but using `/etc/services` would be just a bit cleaner in terms of separating code from data.

Reply:
Ok, let it remain hard-coded. This number is indeed unlikely to ever change.